src/Uno.UI/UI/Xaml/Media/Animation/ColorAnimationUsingKeyFrames.cs

```diff
@@ -388,6 +404,12 @@ protected override void Dispose(bool disposing)
 			base.Dispose(disposing);
 		}
 
+		private protected override void OnThemeChangedCore()
+		{
+			// Value may have changed
+			_finalValue = FindFinalValue() ?? default;
```

Comment:
From my understanding, you are doing this in case that the theme changes while the animation is running. It would mean that the animation will continue as-is and only the final frame will apply the updated value. I think that visual result is going to be really weird, and we should probably just stop animation if the final value has changed (the best solution would be to restart animation from the current value and animate to the updated value, but I think it's probably a lot of work for a stretch case).

Reply:
Indeed, and in fact, that is exactly what uwp is doing: replaying the animation with the updated value. I omitted that part, since I am a bit rushed to get this pr ready. Replaying the animation on theme change deserve to be in a separate pr alone. There is also the fact that, theme changing is quite expensive, and we are unlikely tp see the replay in effect, given that most animations have a really short window.

Reply:
Definitely agree that replaying the animation is most probably not really useful currently. But I would still stop the animation and go directly to the final/updated value. You probably just have to invoke `OnEnd()` (didn't validate the full state machine, but seems to be enough).